pandas/core/nanops.py

```diff
@@ -200,12 +201,88 @@ def _get_fill_value(dtype, fill_value=None, fill_value_typ=None):
                 return tslibs.iNaT
 
 
-def _get_values(values, skipna, fill_value=None, fill_value_typ=None,
-                isfinite=False, copy=True, mask=None):
-    """ utility to get the values view, mask, dtype
-    if necessary copy and mask using the specified fill_value
-    copy = True will force the copy
+def _maybe_get_mask(values: np.ndarray, skipna: bool,
+                    mask: Optional[np.ndarray]) -> Optional[np.ndarray]:
+    """ This function will compute a mask iff it is necessary. Otherwise,
+    return the provided mask (potentially None) when a mask does not need to be
+    computed.
+
+    A mask is never necessary if the values array is of boolean or integer
+    dtypes, as these are incapable of storing NaNs. If passing a NaN-capable
+    dtype that is interpretable as either boolean or integer data (eg,
+    timedelta64), a mask must be provided.
+
+    If the skipna parameter is False, a new mask will not be computed.
+
+    The mask is computed using isna() by default. Setting invert=True selects
+    notna() as the masking function.
+
+    Parameters
+    ----------
+    values :
+        input array to potentially compute mask for
+    skipna :
+        boolean for whether NaNs should be skipped
+    mask :
+        nan-mask if known
+
+    Returns
+    -------
+    Optional[np.ndarray]
+
+    """
+
+    if mask is None:
+        if is_bool_dtype(values.dtype) or is_integer_dtype(values.dtype):
+            # Boolean data cannot contain nulls, so signal via mask being None
+            return None
+
+        if skipna:
+            mask = isna(values)
+
+    return mask
+
+
+def _get_values(values: np.ndarray, skipna: bool, fill_value: Any = None,
+                fill_value_typ: str = None, mask: Optional[np.ndarray] = None
+                ) -> Tuple[np.ndarray, Optional[np.ndarray], np.dtype,
+                           np.dtype, Any]:
+    """ Utility to get the values view, mask, dtype, dtype_max, and fill_value.
+
+    If both mask and fill_value/fill_value_typ are not None and skipna is True,
+    the values array will be copied.
+
+    For input arrays of boolean or integer dtypes, copies will only occur if a
+    precomputed mask, a fill_value/fill_value_typ, and skipna=True are
+    provided.
+
+    Parameters
+    ----------
+    values :
```

Comment:
can you add the types (values : np.array), etc.

Reply:
Done